test/functional/neutronless/test_disconnected_service.py

```diff
@@ -0,0 +1,149 @@
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import mock
+import os
+from pprint import pprint as pp
+import requests
+requests.packages.urllib3.disable_warnings()
+import pytest
+from pytest import symbols
+import time
+
+from oslo_config import cfg
+
+from f5_openstack_agent.lbaasv2.drivers.bigip.icontrol_driver import\
+    iControlDriver
+
+from f5.utils.testutils.registrytools import register_device
+
+osd = os.path.dirname
+DISTRIBUTIONROOT = osd(osd(osd(osd(__file__))))
+del osd
+SERVICELIBDIR = os.path.join(DISTRIBUTIONROOT,
+                             'devtools',
+                             'sample_data',
+                             'service_library')
+LISTENER_ID = u'ffffffff-ffff-ffff-ffff-ffffffffffff'
+CREATELB_SVC =\
+    json.load(open(os.path.join(SERVICELIBDIR, 'createlb.json'), 'r'))
+DISCONNECTED_SVC =\
+    json.load(open(os.path.join(
+        SERVICELIBDIR, 'disconnected.json'), 'r'))['service']
+CONNECTED_SVC =\
+    json.load(open(os.path.join(
+        SERVICELIBDIR, 'connected.json'), 'r'))['service']
+DELETELB_SVC =\
+    json.load(open(os.path.join(SERVICELIBDIR, 'deletelb.json'), 'r'))
+
+pp(DISCONNECTED_SVC)
+DISCONNECTED_SVC['pools'][0]['listeners'][0]['id'] = LISTENER_ID
+DISCONNECTED_SVC['listeners'][0]['id'] = LISTENER_ID
+CONNECTED_SVC['pools'][0]['listeners'][0]['id'] = LISTENER_ID
+CONNECTED_SVC['listeners'][0]['id'] = LISTENER_ID
+for listener in DISCONNECTED_SVC['listeners']:
+    listener['provisioning_status'] = 'PENDING_CREATE'
+for pool in DISCONNECTED_SVC['pools']:
+    pool['provisioning_status'] = 'PENDING_CREATE'
+for monitor in DISCONNECTED_SVC['healthmonitors']:
+    monitor['provisioning_status'] = 'PENDING_CREATE'
+for member in DISCONNECTED_SVC['members']:
+    member['provisioning_status'] = 'PENDING_CREATE'
+    member['network_id'] = u'a8f301b2-b7b9-404a-a746-53c442fa1a4f'
+
+for listener in CONNECTED_SVC['listeners']:
+    listener['provisioning_status'] = 'PENDING_CREATE'
+for pool in CONNECTED_SVC['pools']:
+    pool['provisioning_status'] = 'PENDING_CREATE'
+for monitor in CONNECTED_SVC['healthmonitors']:
+    monitor['provisioning_status'] = 'PENDING_CREATE'
+for member in CONNECTED_SVC['members']:
+    member['provisioning_status'] = 'PENDING_CREATE'
+    member['network_id'] = u'a8f301b2-b7b9-404a-a746-53c442fa1a4f'
+
+
+@pytest.fixture(scope='module')
+def setup_registry_snapshot(bigip):
+    # Setup device registries
+    pp('inside setup_registry_snapshot')
+    return frozenset(register_device(bigip))
+
+
+@pytest.fixture
+def setup_neutronless(request, bigip, setup_registry_snapshot):
```

Comment:
Move this fixture to a common place for all neutronless functional tests to share.